linera-views/src/backends/lru_caching.rs

```diff
@@ -1,45 +1,305 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-//! Add LRU (least recently used) caching to a given store.
+//! Add caching to a given store.
+//! The current policy is based on the LRU (Least Recently Used).
 
-/// The standard cache size used for tests.
-pub const TEST_CACHE_SIZE: usize = 1000;
+use std::{fs::File, io::BufReader};
+
+/// The parametrization of the cache
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct StorageCachePolicy {
+    /// The maximum size of the cache, in bytes (keys size + value sizes)
+    pub max_cache_size: usize,
+    /// The maximum size of an entry size, in bytes
+    pub max_entry_size: usize,
+    /// The maximum number of entries in the cache.
+    pub max_cache_entries: usize,
+}
+
+/// The maximum number of entries in the cache.
+/// If the number of entries in the cache is too large then the underlying maps
+/// become the limiting factor
+pub const DEFAULT_STORAGE_CACHE_POLICY: StorageCachePolicy = StorageCachePolicy {
+    max_cache_size: 10000000,
+    max_entry_size: 1000000,
+    max_cache_entries: 1000,
```

Comment:
This is kindof a duplicate of `max_cache_size`, isn't it? i.e. both limit the size of a cache but in different ways. I think it's possible to set them in a "incompatible" way. For example, with this`DEFAULT`, we have 1 entry having (approximately) ~10Kb. Which is probably a lot 🤔

Reply:
We need several ways to control the cache. They are not all independent of course.
In the chosen settings, we have at most 1M for an individual entry, at most 10M for the full cache and at most 1000 entries in the cache. Of course, if we have N entries and each entry has at most M bytes then the total size is going to be at most N*M.